docs/source/guides/links.rst

```diff
@@ -0,0 +1,89 @@
+Links
+~~~~~
+
+In order to write neural networks, we have to combine functions with *parameters* and optimize the parameters.
+You can use **links** to do this.
+A link is an object that holds parameters (i.e. optimization targets).
+
+The most fundamental ones are links that behave like regular functions while replacing some arguments by their parameters.
+We will introduce higher level links, but here think of links as simply functions with parameters.
+
+One of the most frequently used links is the :class:`~functions.Linear` link (a.k.a. *fully-connected layer* or *affine transformation*).
+It represents a mathematical function :math:`f(x) = Wx + b`, where the matrix :math:`W` and the vector :math:`b` are parameters.
+This link corresponds to its pure counterpart :func:`~functions.linear`, which accepts :math:`x, W, b` as arguments.
+A linear link from three-dimensional space to two-dimensional space is defined by the following line:
+
+.. doctest::
+
+   >>> f = L.Linear(3, 2)
+
+.. note::
+   Most functions and links only accept mini-batch input, where the first dimension of the input array is considered as the *batch dimension*.
+   In the above Linear link case, input must have shape of (N, 3), where N is the mini-batch size.
```

Comment:
Should (N, 3) and N be math expressions? I also noticed that ``(N, M)`` is later used. As long as it's consistent, I think it's fine though.